### PR TITLE
Move configuration to ConfigMap, use kustomize for configuration management

### PR DESCRIPTION
**Pull request by Monkeyanator**
This PR migrates the existing configuration to follow [kustomize](https://github.com/kubernetes-sigs/kustomize) standard patterns. This is needed because users will want to configure the sampling rate and other configuration parameters differently between different environments. The `Makefile` now uses kustomize to generate and pipe in configuration to `kubectl`.

This PR also adds a flag to the webhook server to configure sampling rate, which is passed in through an argument in the `trace-context-injector-args` kustomize patch.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -18,20 +18,16 @@ cluster-down: delete-config delete-certs
...
 delete-certs:
 	@echo 'Deleting mutating controller certs...'
 	kubectl delete secret trace-context-injector-webhook-certs
 
 deploy-config:
 	@echo 'Applying configuration to active cluster...'
-	kubectl apply -f deploy/deployment.yaml
-	kubectl apply -f deploy/mutatingwebhook-ca-bundle.yaml
-	kubectl apply -f deploy/service.yaml
+	kustomize build deploy/overlays/default | kubectl apply -f -
```
Comment (dashpole) [inline Makefile]: generally, the default and the base should be the same.  You can provide example patches for extra functionality to show how users could choose to deviate from the default.

File: deploy/overlays/default/trace-context-injector-args.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+kind: Deployment
+metadata:
+  name: trace-context-injector-webhook-deployment
+spec:
+  template:
+    spec:
+      containers:
+      - name: trace-context-injector
+        args:
+          - -rate=1.0
```
Comment (dashpole) [inline deploy/overlays/default/trace-context-injector-args.yaml]: does this patch actually do anything?  It looks like it just uses the defaults...  A better example patch might be showing how you can decrease the sampling rate to 10% or something.

File: traceutil.go [hunk 3]
```diff
@@ -3,14 +3,26 @@ package main
...
 	"context"
 	"encoding/base64"
+	"errors"
 
 	"go.opencensus.io/trace"
 	"go.opencensus.io/trace/propagation"
 )
 
+// TraceConfig is the struct used to pass configuration to the tracer
+type TraceConfig struct {
```
Comment (dashpole) [inline traceutil.go]: Are we just going to end up redefining `go.ioencensus.io/trace.Config` here?  I'm not convinced we need a new TraceConfig struct.

Generally, we should take all of the inputs (flags, etc), put them in a Config struct, and then define a validate function on that config struct.

Or better yet, [use a ConfigMap](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) to hold this information so we could write patches for individual fields, and don't have to use any flags other than the location of the config file.

File: Makefile (hunk 1 above)
Comment (Monkeyanator) [reply Makefile]: done

File: deploy/overlays/default/trace-context-injector-args.yaml (hunk 2 above)
Comment (Monkeyanator) [reply deploy/overlays/default/trace-context-injector-args.yaml]: done

File: traceutil.go (hunk 3 above)
Comment (Monkeyanator) [reply traceutil.go]: done

File: config.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+	yaml "gopkg.in/yaml.v2"
+)
+
+// DefaultConfigPath refers to location of configuration mount dir specified in deployment
+const DefaultConfigPath = "/etc/webhook/config/config.yaml"
+
+// Config represents all config we need to initialize the webhook server
+type Config struct {
+	Trace struct {
```
Comment (dashpole) [inline config.go]: I think this means the `Trace` struct is embedded in the `Config` struct, meaning you can do config.SampleRate.  Can we instead explicitly name the trace struct subfield, so it is clear what the yaml key is for it?  I.E.
```
type Config struct {
    Trace Trace `yaml:"trace"`
}

// Trace is the configuration for the trace context added to pods
type Trace struct {
    SampleRate float64 `yaml:"sampleRate"`
}
```

**Review by dashpole — COMMENTED** [review]
looks much better.  One small comment, and then this is good to go.

File: config.go (hunk 4 above)
Comment (Monkeyanator) [reply config.go]: done

**Comment by dashpole** [comment]
Related issue: https://github.com/kubernetes-sigs/mutating-trace-admission-controller/issues/3
This is an improvement over the current state.  We should also:

* Include instructions / links on using kustomize
* Make installation possible without running anything locally

/lgtm
/approve

**Pull request merged**